### ISSUE TITLE
Falsey field values result in deletion of form state

### ISSUE THREAD

**Issue by TLadd** [davidkpiano/react-redux-form#110]
Related to the fix for https://github.com/davidkpiano/react-redux-form/issues/84.

Starting in 0.9.3, changing a field value to a falsey value results in the field being deleted from the form state.

https://github.com/davidkpiano/react-redux-form/blob/master/src/reducers/form-reducer.js#L93 is the line causing the issue. From my understanding, this is really looking to be used when actions.remove is called, but since both actions.remove and actions.change are firing off a change action under the hood, the formReducer removes the field data for a field when doing something like actions.change('test.foo', false).

I personally experience the issue with a custom checkbox input that is using true/false for checked/unchecked. When I uncheck the box, the form state is deleted. When I check it, the form state is added back.

It seems like false, null, and even undefined ought to be allowable model value states.

Here is a test case for formReducer that demonstrates the issue I'm having and currently fails:
```
  it('should allow falsey values and maintain form state', () => {
    const reducer = formReducer('test', {
      foo: true,
    });

    const actual = reducer(undefined, actions.change('test.foo', false));

    assert.notEqual(actual.fields.foo, undefined);
  });
```

**Comment by davidkpiano** [comment]
Will be released in the next patch!

**Comment by davidkpiano** [comment]
Fixed in `0.9.6`!
